bot-components/GitHub_types.mli

```diff
@@ -81,7 +83,39 @@ type comment_info =
   ; review_comment: bool
   ; id: id }
 
-type check_run_info = {id: int; node_id: id; url: string}
-
 type push_info =
   {owner: string; repo: string; base_ref: string; commits_msg: string list}
+
+type check_run_status = COMPLETED | IN_PROGRESS | QUEUED
+
+type check_conclusion =
+  | ACTION_REQUIRED
+  | CANCELLED
+  | FAILURE
+  | NEUTRAL
+  | SKIPPED
+  | STALE
+  | SUCCESS
+  | TIMED_OUT
+
+type check_suite_info =
+  {id: int; node_id: id; head_sha: string; url: string; status: check_run_status}
+
+type check_run_info =
+  { id: int
+  ; node_id: id
+  ; head_sha: string
+  ; status: check_run_status
+  ; url: string
+  ; check_suite_info: check_suite_info
+  ; repository_info: repository_info }
+
+type check_run =
+  { id: int
+  ; node_id: id
+  ; head_sha: string
+  ; name: string
+  ; status: string
+  ; url: string
+  ; title: string
+  ; text: string }
```

Comment:
What is the reason for having these two types? Where is `check_run` used?

Reply:
The `check_run` record is used in the `get_check_runs` query. However, a simplification in the code make it no longer relevant. We can remove it if you prefer.

Reply:
No need.